core-rust/state-manager/src/protocol/mainnet_updates/anemone.rs

```diff
@@ -0,0 +1,70 @@
+use crate::flash_templates::consensus_manager_config_flash;
+use crate::query::StateManagerSubstateQueries;
+use crate::{
+    ProtocolUpdateFlashTxnCommitter, ProtocolUpdater, StateComputerConfigurator,
+    StateManagerDatabase, StateUpdateExecutor, ANEMONE_PROTOCOL_VERSION,
+};
+use node_common::locks::StateLock;
+use radix_engine::prelude::dec;
+use radix_engine::utils::generate_seconds_precision_state_updates;
+use radix_engine_common::prelude::NetworkDefinition;
+use std::ops::Deref;
+use std::sync::Arc;
+
+pub struct AnemoneProtocolUpdater {
+    pub network: NetworkDefinition,
+    pub store: Arc<StateLock<StateManagerDatabase>>,
```

Comment:
Feels like to me that an "updater" should not have direct access to some store as it becomes stateful. Rather the store should be passed in to the execute or update function.